member-service/src/main/java/com/hedvig/memberservice/aggregates/MemberAggregate.kt

```diff
@@ -0,0 +1,588 @@
+    constructor(command: CreateMemberCommand) : this() {
+        apply(MemberCreatedEvent(command.memberId, MemberStatus.INITIATED))
+            .andThenApply {
+                if (command.acceptLanguage != null && !command.acceptLanguage.isEmpty()) {
+                    return@andThenApply AcceptLanguageUpdatedEvent(command.memberId, command.acceptLanguage)
+                }
+                null
+            }
+    }
+
+    @CommandHandler
+    fun authAttempt(command: SwedishBankIdAuthenticationAttemptCommand) {
+        var applyChain: ApplyMore? = null
+        val bankIdAuthResponse = command.bankIdAuthResponse
+        if (status == MemberStatus.INITIATED || status == MemberStatus.ONBOARDING) {
+            // Trigger fetching of bisnode data.
+            val ssn = bankIdAuthResponse.sSN
+            applyChain = apply(SSNUpdatedEvent(id, ssn, SSNUpdatedEvent.Nationality.SWEDEN))
+            // -- Tracking id generation for the new member id
+            generateTrackingId()
+            applyChain = getBirthdateFromSwedishSsnAndApplyEvent(ssn, applyChain)
+            applyChain = try {
+                getPersonInformationFromBisnode(applyChain, ssn)
+            } catch (ex: RuntimeException) {
+                logger.error("Caught exception calling bisnode for personalInformation", ex)
+                applyChain!!.andThenApply {
+                    NameUpdatedEvent(
+                        id,
+                        formatName(bankIdAuthResponse.givenName),
+                        formatName(bankIdAuthResponse.surname))
+                }
+            }
+            if (status == MemberStatus.INITIATED) {
+                applyChain = applyChain!!.andThenApply { MemberStartedOnBoardingEvent(id, MemberStatus.ONBOARDING) }
+            }
+        }
+        val authenticatedEvent = MemberAuthenticatedEvent(id, bankIdAuthResponse.referenceToken)
+        if (applyChain != null) {
+            applyChain.andThenApply(Supplier<Any> { authenticatedEvent })
+        } else {
+            apply(authenticatedEvent)
+        }
+    }
+
+    private fun getBirthdateFromSwedishSsnAndApplyEvent(ssn: String?, @Nullable applyChain: ApplyMore?): ApplyMore? {
+        var applyChain = applyChain
+        val birthDate = getBirthdateFromSwedishSsn(ssn)
+        if (birthDate != null) {
+            if (applyChain != null) {
+                applyChain = applyChain.andThenApply(
+                    Supplier<Any> { BirthDateUpdatedEvent(id, birthDate) }
+                )
+                return applyChain
+            }
+            apply(BirthDateUpdatedEvent(id, birthDate))
+        }
+        return applyChain
+    }
+
+    private fun generateTrackingId() {
+        apply(TrackingIdCreatedEvent(id, uuidGenerator.generateRandom()))
+    }
+
+    private fun formatName(name: String): String {
+        val lowercase = name.toLowerCase()
+        return Character.toUpperCase(lowercase[0]).toString() + lowercase.substring(1)
+    }
```

Comment:
If the name consists of multiple words this will make t he following words lower cased

Maybe something like this if we want to handle that:
```suggestion
        private fun formatName(name: String): String {
        return name.toLowerCase().split(" ").map(name -> name[0].toUpper() + name.substring(1)).join(" ")
    }
```

Reply:
I'm fine with both implementations I will stick to the old one